File: oa-portal/problems/m43-campus-shuttle-route/starters/main.cpp
#include <bits/stdc++.h>
using namespace std;

/* dist[i][j] is the distance from location i to location j, 0 being the
   depot. Not symmetric. Return the best round trip. */
long long shortestRoute(const vector<vector<long long>>& dist) {
    // write your code here

    return 0;
}

int main() {
    int n;
    if (scanf("%d", &n) != 1) return 0;
    vector<vector<long long>> dist(n + 1, vector<long long>(n + 1));
    for (int i = 0; i <= n; i++)
        for (int j = 0; j <= n; j++) scanf("%lld", &dist[i][j]);

    printf("%lld\n", shortestRoute(dist));
    return 0;
}

File: oa-portal/problems/m43-campus-shuttle-route/starters/main.java
import java.io.*;
import java.util.*;

public class Main {

    /* dist[i][j] is the distance from location i to location j, 0 being the
       depot. Not symmetric. Return the best round trip. */
    static long shortestRoute(long[][] dist) {
        // write your code here

        return 0;
    }

    public static void main(String[] args) throws IOException {
        FastReader in = new FastReader();
        int n = in.nextInt();
        long[][] dist = new long[n + 1][n + 1];
        for (int i = 0; i <= n; i++)
            for (int j = 0; j <= n; j++) dist[i][j] = in.nextLong();

        System.out.println(shortestRoute(dist));
    }

    static class FastReader {
        private final DataInputStream in = new DataInputStream(new BufferedInputStream(System.in, 1 << 16));
        int nextInt() throws IOException { return (int) nextLong(); }
        long nextLong() throws IOException {
            int b = in.read();
            while (b != '-' && (b < '0' || b > '9')) b = in.read();
            boolean neg = b == '-';
            if (neg) b = in.read();
            long v = 0;
            while (b >= '0' && b <= '9') { v = v * 10 + (b - '0'); b = in.read(); }
            return neg ? -v : v;
        }
        String next() throws IOException {
            int b = in.read();
            while (b <= ' ') b = in.read();
            StringBuilder sb = new StringBuilder();
            while (b > ' ') { sb.append((char) b); b = in.read(); }
            return sb.toString();
        }
    }
}
